comit/src/orderpool.rs

```diff
@@ -509,6 +527,34 @@ mod tests {
         assert_that(&pool.ours().next()).is_none();
     }
 
+    // This test will probably break when expiries config is no longer hard coded
+    // will come from a config file
+    #[test]
+    fn orderpool_filters_out_orders_with_unsafe_expiries() {
+        let mut pool = OrderPool::new(PeerId::random());
+
+        let order_with_safe_expiries = BtcDaiOrder::sell(
+            btc(0.5),
+            dai(9000),
+            SwapProtocol::new(Role::Alice, Position::Sell),
+        );
+
+        let order_with_unsafe_expiries = BtcDaiOrder::sell(btc(0.5), dai(9000), hbit_herc20());
+
+        pool.receive(PeerId::random(), vec![
+            order_with_safe_expiries.clone(),
+            order_with_unsafe_expiries,
+        ]);
+
+        let filtered_orders = pool
+            .safe_expiries()
+            .map(|(_, order)| order.id)
+            .collect::<Vec<OrderId>>();
+
+        assert_eq!(filtered_orders.len(), 1);
```

Comment:
I wanted to matching_does_not_contain here to be more explicity but could not figure out how to do it